### ISSUE TITLE
matplotlib failed to build on 10.11.6

### ISSUE THREAD

**Issue by wedi** [Homebrew/homebrew-python#346]
Hi Community!

Unfortunately installing matplotlib via homebrew fails for me.

`brew install homebrew/python/matplotlib --with-cairo --with-tex`

I double checked and firmly believe that I have installed all requirements and everything is up-to-date.

Here’s the related log-gist: https://gist.github.com/wedi/a5d4c16126b6af5093535e16666f9f4a

Thank you! All help is very much appreciated.

**Comment by tdsmith** [comment]
Please upgrade your CLT to match your Xcode version with `xcode-select --install`.

**Comment by wedi** [comment]
Hi. thanks for your response!

I checked for updates.
`xcode-select —install` gives me:
`xcode-select: error: command line tools are already installed, use "Software Update" to install updates` and there are no software updates offered…

I googled a while and finally found [this thread](https://forums.developer.apple.com/message/180161#180161) and finally this seemingly related issue Homebrew/brew#972

**Comment by tdsmith** [comment]
I see, thanks. I have no advice.

**Comment by tdsmith** [comment]
Though you may have some luck with `sudo xcode-select -s /Library/Developer/CommandLineTools`.

**Comment by ilovezfs** [comment]
> sudo xcode-select -s /Library/Developer/CommandLineTools

That will basically break just about everything.

**Comment by ilovezfs** [comment]
Assuming your Xcode.app is installed at
```
/Applications/Xcode.app
```

You need to run:
```
sudo xcode-select -r
```

Also your Xcode.app is now outdated. So you will have to upgrade to Xcode 8.1 as well as upgrade your command line tools.

**Comment by ilovezfs** [comment]
Closing this for now. Please let us know if it's still a problem after following the above steps.

**Comment by tdsmith** [comment]
@ilovezfs can you tl;dr this situation for me please

**Comment by ilovezfs** [comment]
TLDR is this particular bug was fixed in
https://github.com/Homebrew/homebrew-python/pull/358
https://github.com/Homebrew/homebrew-python/pull/357

However, `sudo xcode-select -s /Library/Developer/CommandLineTools` when you're El Capitan with Xcode 8 and the CLT installed has caused a ton of build failures in Core and we should probably add a doctor check or even hard failure if that configuration is detected.

For instance https://github.com/Homebrew/homebrew-core/issues/5639#issuecomment-252232360 but there are many more.

I think I first noted it here: https://github.com/Homebrew/brew/pull/1001#issuecomment-247840168

**Comment by wedi** [comment]
I can confirm that it works now. Thank you for taking care of this issue! 👍🏻
